src/Cloud4RPi.cpp

```diff
@@ -0,0 +1,199 @@
+#include "Cloud4RPi.h"
+
+const int JSON_BUFFER_SIZE = 500;
+
+Cloud4RPi::Cloud4RPi(const String &_deviceToken, const String &_server, int _port) :
+    deviceToken(_deviceToken),
+    server(_server),
+    port(_port),
+    mqttClient(NULL),
+    jsonBufferSize(JSON_BUFFER_SIZE),
+    variables(new C4RVariableStorage()) {
+}
+
+Cloud4RPi::~Cloud4RPi() {
+    if (mqttClient != NULL) {
+        delete mqttClient;
+        mqttClient = NULL;
+    };
+
+    if (variables != NULL) {
+        delete variables;
+        variables = NULL;
+    }
+}
+void Cloud4RPi::begin(Client& _client) {
+    mqttClient = new PubSubClient(_client);
+    mqttClient->setServer(server.c_str(), port);
+}
+
+bool Cloud4RPi::loop() {
+    return mqttClient->loop();
```

Comment:
Seems like mqttClient could be null here if Cloud4RPi::begin hadn't been called before.

Reply:
Fixed with no throwing an exception.
NOTE: PubSubClient allows to pass NULL-wifi-client to own constructor and handle it in implementation.